toolz/functoolz.py

```diff
@@ -200,6 +200,7 @@ def __init__(self, *args, **kwargs):
 
         self.__doc__ = getattr(func, '__doc__', None)
         self.__name__ = getattr(func, '__name__', '<curry>')
+        self.__module__ = getattr(func, '__module__', None)
```

Comment:
Thanks for the critical eye and expert knowledge, @pitrou!  More tests coming...